app/knight.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.knight = knight
+
+    def preparation(self) -> None:
+        self.name = self.knight["name"]
+        self.power = (self.knight["power"]
+                      + (self.knight["weapon"].get("power", 0)
+                         if self.knight["weapon"] else 0)
+                      + (self.knight["potion"]["effect"].get("power", 0)
+                         if self.knight["potion"] else 0))

```

Comment:
There is a lot of code repetition here. You are checking if `self.knight['weapon']` and `self.knight['potion']` exist twice. Consider creating helper functions for these checks to follow the DRY (Don't Repeat Yourself) principle. This is a code quality issue.